doc/weakly_supervised.rst

```diff
@@ -127,9 +127,16 @@ through the argument `preprocessor` (see below :ref:`fit_ws`)
 Fit, transform, and so on
 -------------------------
 
-The goal of weakly-supervised metric-learning algorithms is to transform
-points in a new space, in which the tuple-wise constraints between points
-are respected.
+Generally, the goal of weakly-supervised metric-learning algorithms is to
+transform points in a new space, in which the tuple-wise constraints between
+points are respected.
+
+But there are also some algorithms that learn a similarity, not a distance,
+thus the points cannot be transformed into a new space. But the goal is the
+same: respect the maximum number of constraints while learning the similarity.
+
+Weakly-supervised mahalanobis learners can transform points into a new space,
+and to do so, we fit the metric learner (example:`MMC`).
```

Comment:
Similar as for supervised learners:

The goal of weakly supervised metric learning algorithms is to learn a (distance or similarity) metric such that the tuple-wise constraints between points are respected.

To do so, we first need to fit the weakly supervised metric learner on a dataset of tuples, as in the example below with `MMC`.